src/Navigation/Plugins/recastJSPlugin.ts

```diff
@@ -0,0 +1,267 @@
+import { INavigationEnginePlugin, ICrowd, AgentParameters, NavMeshParameters } from "../../Navigation/INavigationEngine";
+import { Logger } from "../../Misc/logger";
+import { VertexData } from "../../Meshes/mesh.vertexData";
+import { AbstractMesh } from "../../Meshes/abstractMesh";
+import { Mesh } from "../../Meshes/mesh";
+import { Scene } from "../../scene";
+import { Vector3 } from '../../Maths/math';
+import { TransformNode } from "../../Meshes/transformNode";
+
+declare var Recast: any;
+
+/**
+ * RecastJS navigation plugin
+ */
+export class RecastJSPlugin implements INavigationEnginePlugin {
+    /**
+     * Reference to the Recast library
+     */
+    public bjsRECAST: any = {};
+    public name: string = "RecastJSPlugin";
+    public navMesh: any;
+
+    /**
+     * Initializes the recastJS plugin
+     */
+    public constructor(recastInjection: any = Recast) {
+        if (typeof recastInjection === "function") {
+            recastInjection(this.bjsRECAST);
+        } else {
+            this.bjsRECAST = recastInjection();

```

Comment:
Is this possible? Looks like the usage here is expecting recastInjection to be a function, so this else case can't happen?

Reply:
I did the same way as ammojs. I'm not familiar yet with that. So, any code snippet is welcome :)

Reply:
I didn't look at ammojs, but this doesn't look right to me.

Reply:
Looked at ammojs...
        if (typeof ammoInjection === "function") {
            ammoInjection(this.bjsAMMO);
        } else {
            this.bjsAMMO = ammoInjection;
        }
```

Notice the lack of `()` on the ammoInjection
```